tests/baselines/reference/propertyOrdering.errors.txt

```diff
@@ -1,17 +1,20 @@
+tests/cases/compiler/propertyOrdering.ts(6,23): error TS2301: Initializer of instance member variable '_store' cannot reference identifier 'store' declared in the constructor.
 tests/cases/compiler/propertyOrdering.ts(6,23): error TS2304: Cannot find name 'store'.
 tests/cases/compiler/propertyOrdering.ts(9,34): error TS2551: Property 'store' does not exist on type 'Foo'. Did you mean '_store'?
 tests/cases/compiler/propertyOrdering.ts(16,25): error TS2339: Property '_store' does not exist on type 'Bar'.
 tests/cases/compiler/propertyOrdering.ts(20,14): error TS2339: Property '_store' does not exist on type 'Bar'.
 
 
-==== tests/cases/compiler/propertyOrdering.ts (4 errors) ====
+==== tests/cases/compiler/propertyOrdering.ts (5 errors) ====
     class Foo {
         constructor(store: string) { }
           public foo() {
                 return this._store.length;   // shouldn't be an error
           }
           public _store = store; // no repro if this is first line in class body
                           ~~~~~
+!!! error TS2301: Initializer of instance member variable '_store' cannot reference identifier 'store' declared in the constructor.
+                          ~~~~~
 !!! error TS2304: Cannot find name 'store'.

```

Comment:
Weird that you get both errors now.

Reply:
I'm suspecting that the new error is the expected one and the "cannot find name" isn't? At least that's how I understand this comment:
https://github.com/AlCalzone/TypeScript/blob/1213c35d57d91cc773f05c3e5dc36770c602ba94/src/compiler/checker.ts#L1956-L1973

Reply:
Well...yeah I suppose if we report the new one, the old one should disappear - OR this should be an elaboration on the `Cannot find name` one.

But why does this change even make this error appear?

Reply:
Any hint how to find that out? I'm not familiar with the TS codebase.

Reply:
The way I'd do it would be to breakpoint on the line that adds that diagnostic, and see who's calling it. Maybe this spelling stuff is being used more than it should?

Reply:
I think the issue is that this case comes into effect and does not set the `result` variable:
![grafik](https://user-images.githubusercontent.com/17641229/176117631-868c1a2b-4122-4d55-954c-2e1fefe119a2.png)

This means that this section of the code is called, reporting the "cannot find store" error in the currently paused line:
![grafik](https://user-images.githubusercontent.com/17641229/176117866-245d35a3-8ca3-4ad7-9816-70585a0316f9.png)

For reference, this is the top of the call stack:
```
<anonymous> (i:\TypeScript\src\compiler\checker.ts:2205)
addLazyDiagnostic (i:\TypeScript\src\compiler\checker.ts:41845)
resolveNameHelper (i:\TypeScript\src\compiler\checker.ts:2166)
resolveName (i:\TypeScript\src\compiler\checker.ts:1817)
getResolvedSymbol (i:\TypeScript\src\compiler\checker.ts:23216)
checkIdentifier (i:\TypeScript\src\compiler\checker.ts:25708)
checkExpressionWorker (i:\TypeScript\src\compiler\checker.ts:34904)
checkExpression (i:\TypeScript\src\compiler\checker.ts:34847)
checkExpressionCached (i:\TypeScript\src\compiler\checker.ts:34484)
checkDeclarationInitializer (i:\TypeScript\src\compiler\checker.ts:34507)
getTypeForVariableLikeDeclaration (i:\TypeScript\src\compiler\checker.ts:8985)
getWidenedTypeForVariableLikeDeclaration (i:\TypeScript\src\compiler\checker.ts:9484)
getTypeOfVariableOrParameterOrPropertyWorker (i:\TypeScript\src\compiler\checker.ts:9651)
getTypeOfVariableOrParameterOrProperty (i:\TypeScript\src\compiler\checker.ts:9539)
getTypeOfSymbol (i:\TypeScript\src\compiler\checker.ts:9932)
checkPropertyAccessExpressionOrQualifiedName (i:\TypeScript\src\compiler\checker.ts:29171)
checkPropertyAccessExpression (i:\TypeScript\src\compiler\checker.ts:28913)
checkExpressionWorker (i:\TypeScript\src\compiler\checker.ts:34938)
checkExpression (i:\TypeScript\src\compiler\checker.ts:34847)
checkNonNullExpression (i:\TypeScript\src\compiler\checker.ts:28854)
checkPropertyAccessExpression (i:\TypeScript\src\compiler\checker.ts:28913)
checkExpressionWorker (i:\TypeScript\src\compiler\checker.ts:34938)
checkExpression (i:\TypeScript\src\compiler\checker.ts:34847)
checkExpressionCached (i:\TypeScript\src\compiler\checker.ts:34484)
```

Reply:
To be honest I have no clue what the best mitigation is here.

Reply:
I tried simply reordering the two checks which gets rid of the double error here but adds some bad changes elsewhere. See https://github.com/microsoft/TypeScript/pull/49575/commits/6b521325fb3aec4cc3ac8c7bf3dd685b8b9c85f2 for reference.